DEVELOPER_GUIDE.md

```diff
@@ -27,6 +30,50 @@ changes to Fluent Bit.
     - [Valgrind](#valgrind)
   - [Need more help?](#need-more-help)
 
+## Development Environment
+
+Development environments allow to work without any dependency locally.
+
+### Devcontainer
+
+A Development Container (or Dev Container for short) allows you to use a container as a full-featured development environment.
+
+Thanks to [Visual Studio Code integration](https://code.visualstudio.com/docs/devcontainers/containers), devcontainers are managed easily.
+
+#### Prerequirements
+- [docker](https://www.docker.com)
+- [visual studio code](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) [optional]
+- [devcontainer](https://github.com/devcontainers/cli) [optional]
+
+#### Using without optional tools
+
+```shell
+docker run \
+    --name devcontainer-fluent-bit \
+    --publish 2020:2020 \
+    --volume $PWD/:/workspaces/fluent-bit \
+    --tty \
+    --detach \
+    mcr.microsoft.com/devcontainers/base:jammy
+docker exec -it --user vscode devcontainer-fluent-bit bash
+cd /workspaces/fluent-bit
+sudo bash .devcontainer/install-dev-tools.sh
```

Comment:
Could we just use the debug image?

Reply:
I think we need to add a user which has 1000 uid for better experience. I will try it

Reply:
I updated PR. it works :rocket: